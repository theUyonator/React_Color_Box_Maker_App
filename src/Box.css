.Box {
    padding: 5px;
    margin: 15px;
    border: 1px solid black
  }

.Box-btn {
    padding: 2px;
    margin: 5px;
    background-color: red;
    color: white;
    text-align: center;
}