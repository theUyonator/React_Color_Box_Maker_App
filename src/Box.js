import React from "react";
import "./Box.css"

const Box = ({ id, width, height, backgroundColor, remove }) => {
   return ( 
            <>
                <div 
                    className="Box" 
                    style={{
                        width: `${width}em`,
                        height: `${height}em`,
                        backgroundColor: backgroundColor
                    }}>
                    
                 </div>
                <button 
                    className="Box-btn" 
                    onClick={remove}>X</button>
            </>
   )
}

Box.defaultProps = {
    backgroundColor: "yellow",
    width: 10,
    height: 10
};

export default Box;