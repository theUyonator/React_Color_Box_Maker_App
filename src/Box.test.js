import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Box from "./Box";

describe("<Box /> rendering", () => {

it("renders correctly without crashing", () => {
    render(<Box />)
});

it("matches snapshot", () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});

});



