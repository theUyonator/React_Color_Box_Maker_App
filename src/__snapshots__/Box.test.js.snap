// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Box /> rendering matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="width: 10em; height: 10em; background-color: yellow;"
  />
  <button
    class="Box-btn"
  >
    X
  </button>
</DocumentFragment>
`;
